File: myauthserver.rb
#!/usr/bin/env ruby1.9

require 'thrift'
$:.push('gen-rb')

require 'authentication'
require 'myauth_constants'

class AuthenticationHandler
  def say_hello
    puts "thrift client connected"
    "hello thrift client"
  end

  def login cred
    if cred.username == 'hello' && cred.password == 'world'
      puts "logged in"
      return MyAuth::LoginStatus::SUCCESS 
    end

    puts "great pie of fail"
    MyAuth::LoginStatus::FAIL
  end
end

handler = AuthenticationHandler.new
processor = MyAuth::Authentication::Processor.new(handler)
transport = Thrift::ServerSocket.new(9090)
transportFactory = Thrift::BufferedTransportFactory.new()
server = Thrift::SimpleServer.new(processor, transport, transportFactory)

puts "Starting the server..."
server.serve()
puts "done."

File: myauthserver.py
import sys
sys.path.append('gen-py')

from myauth import Authentication
from myauth.constants import *

from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol
from thrift.server   import TServer

class AuthenticationHandler:
  def __init__(self):
    pass

  def say_hello(self):
    print("thrift client connected")
    return "hello thrift client"

  def login(self, cred):
    if cred.username == 'hello' and cred.password == 'world':
      print("logged in")
      return LoginStatus.SUCCESS

    print("great pie of fail")
    return LoginStatus.FAIL


if __name__ == '__main__':
  handler = AuthenticationHandler()
  processor = Authentication.Processor(handler)
  transport = TSocket.TServerSocket(port=9090)
  tfactory = TTransport.TBufferedTransportFactory()  
  pfactory = TBinaryProtocol.TBinaryProtocolFactory()  
  server = TServer.TSimpleServer(processor, transport, tfactory, pfactory)  
  print 'Starting the server...'  
  server.serve()  
